llvm/lib/Target/PowerPC/PPCInstrVSX.td

```diff
@@ -2731,6 +2731,8 @@ def : Pat<(f32 (fneg (fabs f32:$S))),
                (COPY_TO_REGCLASS $S, VSFRC)), VSSRC))>;
 
 // f32 Min.
+def : Pat<(f32 (fminnum f32:$A, f32:$B)),
+          (f32 FpMinMax.F32Min)>;
```

Comment:
I'd expect a more direct replacement, removing the legality of the IEEE variants. At least should avoid duplicating the patterns

Reply:
We need some steps to remove IEEE variants:
   * Support fminnum/fmaxnum on all architectures with fminnum_ieee/fmaxnum_ieee.
   * Remove all the references of FMAXNUM_IEEE/FMINNUM_IEEE in the common code
   * Remove fminnum_ieee/fmaxnum_ieee in the code of backend.

Reply:
This is a patch for the first step.

Reply:
I don't see a need to have a target have coexisting support for both fminnum and fminnum_ieee. We can fully migrate the target to the new behavior in one go

Reply:
z-turn? 

> But we need sometime to switch FMAXNUM_IEEE to FMAXNUM

Yes, but you can get there by removing the use a target at a time. Here if you make FMAXNUM legal, and stop making FMAXNUM_IEEE legal, the target starts interpreting the nodes in the new way and stops using FMAXNUM_IEEE.

Reply:
> z-turn?
> 

I mean we will need to change the testcases, and then will need to change them back.

> > But we need sometime to switch FMAXNUM_IEEE to FMAXNUM
> 
> Yes, but you can get there by removing the use a target at a time. Here if you make FMAXNUM legal, and stop making FMAXNUM_IEEE legal, the target starts interpreting the nodes in the new way and stops using FMAXNUM_IEEE.

Yes. But we haven't change the code in arch-indepent code.
I will have a try with PowerPC/VSX.

Reply:
I have a try. If we drop FMAXNUM_IEEE/FMINNUM_IEEE now,
`llvm/test/CodeGen/PowerPC/fminimum-fmaximum.ll` will not emit correct instructions, as
`expandFMINIMUM_FMAXIMUM` uses FMINNUM_IEEE/FMAXNUM_IEEE.